hooks/useSignTransaction.ts

```diff
@@ -0,0 +1,45 @@
+import useConnectedWallet from "./useConnectedWallet";
+import { Address, createWalletClient, custom } from "viem";
+import { CHAIN_ID } from "@/lib/consts";
+import getViemNetwork from "@/lib/viem/getViemNetwork";
+import { useFrameProvider } from "@/providers/FrameProvider";
+import { useSwitchChain, useWriteContract } from "wagmi";
+
+type Parameters = {
+  address: Address;
+  account: Address;
+  abi: any;
+  functionName: string;
+  args: any;
+  value: any;
+};
```

Comment:
Would it be possible to use the existing type from `wagmi`?
<img width="776" alt="Screenshot 2025-04-11 at 3 32 03 PM" src="https://github.com/user-attachments/assets/212ade3b-bd9a-4d88-8a16-7620e4e53b26" />

Reply:
https://wagmi.sh/react/api/hooks/useWriteContract#parameters